src/sync/list.rs

```diff
@@ -213,7 +239,7 @@ impl<'g, T: 'g, C: Container<T>> Iterator for Iter<'g, T, C> {
                 let succ = succ.with_tag(0);
 
                 match self.pred.compare_and_set_weak(
-                    self.curr,
+                    self.curr.with_tag(0),
```

Comment:
What if we immediately return `IterError::Stalled` when `self.curr.tag() != 0`?  Because nonzero `self.curr.tag()` means the node containing `self.pred` is deleted, and the iteration is stalled.